### PR TITLE
Wketchum update channel map wib eth

### PR DESCRIPTION
**Pull request by wesketchum**

### CODE REVIEW

**Review by mroda88 — APPROVED** [review]
LGTM

File: include/detchannelmaps/TPCChannelMap.hpp [hunk 1]
```diff
@@ -67,11 +67,10 @@ class TPCChannelMap
   virtual uint get_offline_channel_from_crate_slot_stream_chan(uint crate, uint slot, uint stream, uint channel) {
     
     constexpr uint n_chan_per_stream = 64;
-    constexpr uint n_streams_per_link = 4;
 
-    uint link = (stream >> 6);
-    uint substream = (stream & 0x3f);
-    uint ch = n_chan_per_stream*substream+channel;
+    uint link = (stream >> 6) & 1;
```
Comment (alessandrothea) [inline include/detchannelmaps/TPCChannelMap.hpp:79]: Without a strict argument check, these reduced masks are likely to hide issues with out-of-range arguments. Phr could either use the full bit ,ask for the link/substream fields or add a strict argument check

**Review by alessandrothea — CHANGES_REQUESTED** [review]

**Review by bieryAtFnal — APPROVED** [review]
I see a change in the channel and plane numbers when I use the changes on this branch, but I'm not 100% sure that they are what I should expect.

Before updating, I see output like the following:

```
	====WIB DATA====
		ch 9120 (plane 1): ped = 8981.67, rms = 28.6857
		ch 9119 (plane 1): ped = 9010.64, rms = 30.5870
		ch 9118 (plane 1): ped = 9028.90, rms = 29.9912
		ch 9117 (plane 1): ped = 9087.78, rms = 30.8457
		ch 9116 (plane 1): ped = 9079.71, rms = 30.8230
		ch 9115 (plane 1): ped = 9020.63, rms = 29.8985
		ch 9114 (plane 1): ped = 8859.24, rms = 28.7877
		ch 9113 (plane 1): ped = 9009.19, rms = 28.3105
		ch 9112 (plane 1): ped = 8865.56, rms = 27.7649
		ch 9111 (plane 1): ped = 8847.37, rms = 29.2819
		ch 9110 (plane 1): ped = 1101.28, rms = 24.0471
		ch 9109 (plane 1): ped = 1028.45, rms = 23.8143
		ch 9108 (plane 1): ped = 958.89, rms = 24.5768
		ch 9107 (plane 1): ped = 1039.98, rms = 24.9647
		ch 9106 (plane 1): ped = 1002.24, rms = 26.0746
		ch 9105 (plane 1): ped = 1033.47, rms = 26.3744
		ch 9856 (plane 2): ped = 1312.66, rms = 24.8614
		ch 9857 (plane 2): ped = 993.10, rms = 25.3099
		ch 9858 (plane 2): ped = 953.95, rms = 24.2897
		ch 9859 (plane 2): ped = 1076.50, rms = 24.5854
		ch 9860 (plane 2): ped = 1078.33, rms = 24.3765
		ch 9861 (plane 2): ped = 1060.95, rms = 23.8326
		ch 9862 (plane 2): ped = 8577.14, rms = 29.3228
		ch 9863 (plane 2): ped = 8971.96, rms = 28.1865
```

After updating, I see output like the following:

```
====WIB DATA====
		ch 8136 (plane 0): ped = 8981.67, rms = 28.6857
		ch 8134 (plane 0): ped = 9010.64, rms = 30.5870
		ch 8132 (plane 0): ped = 9028.90, rms = 29.9912
		ch 8130 (plane 0): ped = 9087.78, rms = 30.8457
		ch 8128 (plane 0): ped = 9079.71, rms = 30.8230
		ch 9119 (plane 1): ped = 9020.63, rms = 29.8985
		ch 9121 (plane 1): ped = 8859.24, rms = 28.7877
		ch 9123 (plane 1): ped = 9009.19, rms = 28.3105
		ch 9125 (plane 1): ped = 8865.56, rms = 27.7649
		ch 9127 (plane 1): ped = 8847.37, rms = 29.2819
		ch 9890 (plane 2): ped = 1101.28, rms = 24.0471
		ch 9888 (plane 2): ped = 1028.45, rms = 23.8143
		ch 9886 (plane 2): ped = 958.89, rms = 24.5768
		ch 9884 (plane 2): ped = 1039.98, rms = 24.9647
		ch 9882 (plane 2): ped = 1002.24, rms = 26.0746
		ch 9880 (plane 2): ped = 1033.47, rms = 26.3744
		ch 9881 (plane 2): ped = 1312.66, rms = 24.8614
		ch 9883 (plane 2): ped = 993.10, rms = 25.3099
		ch 9885 (plane 2): ped = 953.95, rms = 24.2897
		ch 9887 (plane 2): ped = 1076.50, rms = 24.5854
		ch 9889 (plane 2): ped = 1078.33, rms = 24.3765
		ch 9891 (plane 2): ped = 1060.95, rms = 23.8326
```

The latter output seems to make more sense, in that the large difference in pedestal values correlates with different planes.  The former output doesn't have any such correlation.

Here is the command that I used:
```
cd /data2 # np04-srv-002
wibdecoder.py -n 1 --print-headers --print-adc-stats --channel-map PD2HDChannelMap --check-timestamps np04_hd_run022577_0010_dataflow0_datawriter_0_20230809T151620.hdf5.copied
```

File: include/detchannelmaps/TPCChannelMap.hpp (hunk 1 above)
Comment (wesketchum) [reply include/detchannelmaps/TPCChannelMap.hpp:79]: Added explicit check on stream number. I think the better way would be to actually use this stream number in the text file maps, but that will be another iteration with offline.

**Review by alessandrothea — APPROVED** [review]

**Pull request merged**